ZelBack/src/services/IOUtils.js

```diff
@@ -0,0 +1,497 @@
+  try {
+    const response = await axios.get(url, { responseType: 'arraybuffer' });
+    const fileData = Buffer.from(response.data, 'binary');
+    if (rename === true) {
+      await fs.writeFile(`${localpath}/backup_${component}.tar.gz`, fileData);
+    } else {
+      const fileNameArray = url.split('/');
+      const fileName = fileNameArray[fileNameArray.length - 1];
+      await fs.writeFile(`${localpath}/${fileName}`, fileData);
+    }
+    return true;
+  } catch (err) {
+    log.error(err);
+    return false;
+  }
+}
+
+/**
+ * Extracts the contents of a tarball (tar.gz) file to the specified extraction path.
+ *
+ * @param {string} extractPath - The path where the contents of the tarball will be extracted.
+ * @param {string} tarFilePath - The path of the tarball (tar.gz) file to be extracted.
+ * @returns {boolean} - True if the extraction is successful, false on failure.
+ */
+async function untarFile(extractPath, tarFilePath) {
+  try {
+    await fs.mkdir(extractPath, { recursive: true });
+    const unpackCmd = `sudo tar -xvzf ${tarFilePath} -C ${extractPath}`;
+    await exec(unpackCmd, { maxBuffer: 1024 * 1024 * 10 });
+    return { status: true };
+  } catch (error) {
+    const stringstderr = error.stderr.replace(/\n/g, ' ');
+    const stringstdout = error.stdout.replace(/\n/g, ' ');
+    log.error('Error during extraction:', error.stderr || error.stdout);
+    return { status: false, error: stringstderr || stringstdout };
+  }
+}
+
+/**
+ * Creates a tarball (tar.gz) archive from the specified source directory.
+ *
+ * @param {string} sourceDirectory - The path of the directory to be archived.
+ * @param {string} outputFileName - The name of the tarball archive file to be created.
+ * @returns {boolean} - True if the tarball is successfully created, false on failure.
+ */
+async function createTarGz(sourceDirectory, outputFileName) {
+  try {
+    const outputDirectory = outputFileName.substring(0, outputFileName.lastIndexOf('/'));
+    await fs.mkdir(outputDirectory, { recursive: true });
+    const packCmd = `sudo tar -czvf ${outputFileName} -C ${sourceDirectory} .`;
+    await exec(packCmd, { maxBuffer: 1024 * 1024 * 10 });
+    return { status: true };
+  } catch (error) {
+    const stringstderr = error.stderr.replace(/\n/g, ' ');
+    const stringstdout = error.stdout.replace(/\n/g, ' ');
+    log.error('Error creating tarball:', error.stderr || error.stdout);
+    return { status: false, error: stringstderr || stringstdout };
+  }
+}
+
+/**
+ * Removes the specified directory and its contents or only the contents.
+ *
+ * @param {string} rpath - The path of the directory to be removed.
+ * @param {boolean} directory - Flag indicating whether to remove only the directory contents (true) or the entire directory (false).
+ * @returns {boolean} - True if the directory or its contents are removed successfully, false on failure.
+ */
+async function removeDirectory(rpath, directory = false) {
```

Comment:
why not using fs module?

Reply:
fs.promises.rmdir will not remove not empty dir i want give ability to remove directory with files and others directory